operator/internal/manifests/openshift/configure.go

```diff
@@ -104,6 +108,89 @@ func ConfigureGatewayDeployment(
 		return kverrors.Wrap(err, "failed to merge sidecar container spec ")
 	}
 
+	// Remove --web.healthchecks.url previous value.
+	argI := -1
+	args := d.Spec.Template.Spec.Containers[gwIndex].Args
+
+	for i, arg := range args {
+		if strings.HasPrefix(arg, "--web.healthchecks.url=") {
+			argI = i
+			break
+		}
+	}
+
+	if argI != -1 {
+		args[argI] = args[len(args)-1]
+		d.Spec.Template.Spec.Containers[gwIndex].Args = args[:len(args)-1]
+	}
```

Comment:
I suggest to simplify this by replacing the value instead of remove/add-anew like this:

```go
for i, arg := range gwArgs {
	if strings.HasPrefix(arg, "--web.healthchecks.url=") {
		gwContainer.Args[i] = "YOUR_NEW_VALUE_HERE"
	}
}
```
WDYT?

Reply:
that works too, but I did it this way to keep the mergo.Merge pattern. Either way is fine by me :)

Reply:
You can still keep the `mergo.Merge` pattern. You first compile the args list and then pass it to the container. Finally you `mergo.Merge`

Reply:
Sure, that's what I do here, but before doing that I delete the `--web.healthchecks.url=` entry from the args list, since doing `mergo.Merge` would just append it and not override it (ending up with two values `--web.healthchecks.url=`). 

Do you suggest that I override this value in the args directly, then compile the rest of the newer args without `--web.healthchecks.url=`? Because that works as well, but it wouldn't follow the `mergo.Merge` since we edit the `--web.healthchecks.url=` value directly instead of doing it via `mergo.Merge`. 
That is the only reason I'm deleting then re inserting the value here, otherwise I can just do it as you suggested.

Reply:
I have a similar feeling to Peri regarding constructing the args and the multiple use of mergo. I think you can re-use the existing `gw*` variables to your advantage and avoid using multiple mergo-calls. I have [a suggestion](https://github.com/aminesnow/loki/compare/gateway-route-https...xperimental:smaller-configure) how this could look.